### PR TITLE
[GEOS-10352] GetMap clip vendor does not always return fully transparent images outside of clip

### PR DESCRIPTION
**Pull request by aaime**
[GEOS-10352](https://osgeo-org.atlassian.net/browse/GEOS-10352)

Set ticket for details.
Depends on https://github.com/geotools/geotools/pull/3731

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [x] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/wms/src/main/java/org/geoserver/wms/map/DirectRasterRenderer.java [hunk 1]
```diff
@@ -398,6 +402,7 @@ public RenderedImage render() throws FactoryException {
                 if (transparent && !image.getColorModel().hasAlpha()) {
                     image = addAlphaChannel(image);
                     worker.setImage(image);
+                    transparencyType = image.getColorModel().getTransferType();
```
Comment (dromagnoli) [inline src/wms/src/main/java/org/geoserver/wms/map/DirectRasterRenderer.java:405]: I'm wondering if that migth be an auto-complete issue.
Is actually getTransparencyType or is getTransferType correct, instead?

**Review by dromagnoli — COMMENTED** [review]
See my comment about a doubt on a getter with probably a wrong name (maybe just a TYPO due to a autocomplete suggestion).

File: src/wms/src/main/java/org/geoserver/wms/map/DirectRasterRenderer.java (hunk 1 above)
Comment (aaime) [reply src/wms/src/main/java/org/geoserver/wms/map/DirectRasterRenderer.java:405]: Auto-complete issue, well spotted.

Comment (dromagnoli) [reply src/wms/src/main/java/org/geoserver/wms/map/DirectRasterRenderer.java:405]: Cool. beside of that, +1 to merge

**Pull request merged**